cli/flox-rust-sdk/src/providers/nix.rs

```diff
@@ -0,0 +1,31 @@
+use std::path::PathBuf;
+use std::process::Command;
+use std::sync::LazyLock;
+
+static NIX_BIN: LazyLock<PathBuf> = LazyLock::new(|| {
+    std::env::var("NIX_BIN")
+        .unwrap_or_else(|_| env!("NIX_BIN").to_string())
+        .into()
+});
+
+/// Returns a `Command` for `nix` with a default set of features enabled.
+pub fn nix_base_command() -> Command {
+    let mut command = Command::new(&*NIX_BIN);
+    command.args([
+        "--option",
+        "extra-experimental-features",
+        "nix-command flakes",
+    ]);
+    command
+}
+
+pub mod test_helpers {
```

Comment:
in d75c062 i made the `#[cfg(any(test, feature = "tests"))]` annotation actually work, so that these modules are not compiled into production binaries, but still available if the "tests" feature is used.

We (used to -- apparently) have that annotation in a few other places, and could reintroduce this now that it actually works.

Reply:
I'm going to punt on this for now because we're not consistently using the pattern for other `test_helpers` and this is the simplest of them all.